### PR TITLE
Simplified the remove()

### PR DESCRIPTION
**Pull request by boqian2000**
Simplified the remove().  I feel the new implementation is more elegant (cutting lines of code from 32 to 18) and conceptually more straightforward.

### CODE REVIEW

File: Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift [hunk 1]
```diff
@@ -350,24 +322,23 @@ extension BinarySearchTree: CustomStringConvertible {
...
+   public func toArray() -> [T] {
+      return map { $0 }
+   }
 
-  public func toArray() -> [T] {
-    return map { $0 }
-  }
 }
+
+//extension BinarySearchTree: CustomDebugStringConvertible {
```
Comment (boqian2000) [inline Binary Search Tree/Solution 1/BinarySearchTree.playground/Sources/BinarySearchTree.swift]: That was a mistake. I didn't intend to check it in.  However, I do feel the description is much more readable than the debugDescription.  It makes the playground much easier and fun to play with. So I still would like to recommend removing CustomDebugStringConvertible().

**Comment by kelvinlauKL** [comment]
@boqian2000 Could you update the `README` file with the updated logic too? 

I agree with your deletion on `CustomDebugStringConvertible` - initially I thought you deleted `CustomStringConvertible`. 

Your change feels good, and the previous implementation really felt clunky.

**Comment by boqian2000** [comment]
Done updating the README file.

**Comment by kelvinlauKL** [comment]
Merged. Thanks @boqian2000

**Pull request merged**